modin/pandas/general.py

```diff
@@ -404,27 +404,35 @@ def concat(
                 sort=sort,
             )
         )
-    if join not in ["inner", "outer"]:
+    if join == "outer":
+        # Filter out empties
+        list_of_objs = [
+            obj
+            for obj in list_of_objs
+            if (
+                isinstance(obj, (Series, pandas.Series))
```

Comment:
I'm still not sure if a separate condition for Series is needed here. We could continue the [discussion](https://github.com/modin-project/modin/pull/5673/files#r1116179037) here.

A small summary: you claim that pandas does not filter empty series at all. I believe that empty series are not filtered only in the case when all objects for concatenation are Series according to https://github.com/pandas-dev/pandas/blob/main/pandas/core/reshape/concat.py#L476.

Reply:
If I remove this condition, the following simple test:
```python
def test_concat_empty_df_series():
    pdf = pandas.concat((pandas.DataFrame({"A": [1, 2, 3]}), pandas.Series()))
    mdf = pd.concat((pd.DataFrame({"A": [1, 2, 3]}), pd.Series()))
    df_equals(pdf, mdf)
```
fails with
```
E           AssertionError: DataFrame are different
E
E           DataFrame shape mismatch
E           [left]:  (3, 2)
E           [right]: (3, 1)
```

Reply:
I've added the test to the PR.

Reply:
I see, thanks